### DISCUSSION TITLE
Refresh specific folder node content

### DISCUSSION THREAD

**Discussion by budgiedownunder** [Studio-42/elFinder#3356] [Q&A]
Is there a client-side API call that can trigger a refresh of a specific tree node (its hash is known) within the nav tree without having to perform a full reload/refresh of the entire tree whilst, at the same time, maintaining the current working directory (which may, or may not, be within target tree)? If not, what would be the best approach to achieve this?

Thanks in advance.

**Comment by nao-pon** [comment]
@budgiedownunder Is it not necessary to update the Tree UI, just update the folder data?

If so, it is:
`fm` is an instance of elFinder.

```javascript
fm.request({
	data: {cmd: 'open', target: 'TARGET_HASH'},
	preventDefault: true,
	asNotOpen: true
}).done(function(data) {
	fm.updateCache(data);
});
```

**Reply by budgiedownunder** [reply]
@nao-pon  - thank you very much for your reply.

Ideally, I would like the tree UI to update as well. Would the approach you suggest trigger the view to update or would I need to add additional steps for that?

For example, if we assume the following in the current view:

```
|- A
|  |- A1
|  |- A2
|
|- B
   |- B1 
   |- B2
```
with B as the CWD. If I were to trigger the refresh request as you have it on the hash associated with A (which would then bring in nodes A1, A2 + a new node A3 into the cache) , would you expect the view to update automatically to then show A1, A2, A3 or would I need to perform additional steps such as triggering a select/open A before then returning the CWD to B?
